client/README.md

```diff
@@ -1,21 +1,168 @@
-# Zecale client
+# Python client to interact with the Zecale aggregator
 
-## Prerequisites
+## Structure of the directory
 
-- [Python3](https://www.python.org/downloads/) (at least version `3.7`)
-- [Pip](https://pip.pypa.io/en/stable/) (at least version `19.0.2`)
+### `zecale`
+
+```
+zecale
+ |_ api
+ |_ cli
+ |_ core
+ |_ dummy_app
+```
+
+This directory contains the API code for the Zecale client (`api`), its backend implementation (`core`), the code for the client CLI (`cli`), and the code of an helper CLI for a "dummy application" (`dummy_app`).
+
+### `test_commands`
+
+This directory contains a list of useful commands to help run the tests, as well as some minimal testing scenarios acting as integration tests.
+
+### `tests`
+
+The `tests` folder contains the unit tests of the `zecale` package.
 
 ## Setup
 
+Ensure that the following are installed:
+
+- Python 3.7 (See `python --version`)
+- [venv](https://docs.python.org/3/library/venv.html#module-venv) module.
+- gcc
+
+First, make sure to setup the Zeth dependency in order to generate the
+API files and be able to use the Zeth CLIs
 ```console
-$ cd client
+$ cd $ZECALE/depends/zeth/client
 $ python -m venv env
-$ . env/bin/activate
-$ make setup
+$ source env/bin/activate
+(env)$ make setup
+```
+
+Then, Execute the following inside the `client` directory.
+```console
+$ cd $ZECALE/client
+$ python -m venv env
+$ source env/bin/activate
+(env)$ make setup
+```
+
+(It may also be necessary to install solc manually if the `py-solc-x` package
+fails to find it. See the instructions below.)
+
+We assume all further commands described here are executed from within the
+Python virtualenv.  To enter the virtualenv from a new terminal, re-run
+```console
+$ source env/bin/activate
+```
+
+## Execute unit tests
+
+```console
+(env)$ make check
+```
+
+## Launch a local Ethereum testnet
+
+In order for Zecale to work, it is necessary for the blockchain to support
+BW6-761 arithmetic. As such, we assume below that the smart contracts are
+deployed on an Ethereum testnet that supports BW6-761 precompiled contracts.
+
+To do so, please follow the instructions below to run a modified version
+of ganache-cli in a new terminal:
+```console
+cd ../depends/zeth/zeth_contracts
+npm install # make sure that node and npm have the right versions
+npm run testrpc
+```
+
```

Comment:
Could potentially refer to the zeth README where we give versions and instructions.

Reply:
Good idea, see: https://github.com/clearmatics/zecale/pull/50/commits/4eb79525ac8c227bfbed3fcc79793bfeb34aa60d